[Bug]: Qwen3-TTS code predictor crashes with "Current vLLM config is not set" during KV cache init

### Your current environment

<details>
<summary>The output of <code>python collect_env.py</code></summary>

```text
Collecting environment information...
==============================
        System Info
==============================
OS                           : Ubuntu 24.04.3 LTS (x86_64)
GCC version                  : (Ubuntu 13.3.0-6ubuntu2~24.04.1) 13.3.0
Clang version                : 18.1.3 (1ubuntu1)
CMake version                : version 3.28.3
Libc version                 : glibc-2.39

==============================
       PyTorch Info
==============================
PyTorch version              : 2.10.0+cu128
Is debug build               : False
CUDA used to build PyTorch   : 12.8
ROCM used to build PyTorch   : N/A

==============================
      Python Environment
==============================
Python version               : 3.12.3 (main, Jan 22 2026, 20:57:42) [GCC 13.3.0] (64-bit runtime)
Python platform              : Linux-6.8.0-90-generic-x86_64-with-glibc2.39

==============================
       CUDA / GPU Info
==============================
Is CUDA available            : True
CUDA runtime version         : 13.0.88
CUDA_MODULE_LOADING set to   : 
GPU models and configuration : 
GPU 0: NVIDIA B200
GPU 1: NVIDIA B200
GPU 2: NVIDIA B200
GPU 3: NVIDIA B200
GPU 4: NVIDIA B200
GPU 5: NVIDIA B200
GPU 6: NVIDIA B200
GPU 7: NVIDIA B200

Nvidia driver version        : 580.126.09
cuDNN version                : Could not collect
HIP runtime version          : N/A
MIOpen runtime version       : N/A
Is XNNPACK available         : True

==============================
          CPU Info
==============================
Architecture:                         x86_64
CPU op-mode(s):                       32-bit, 64-bit
Address sizes:                        52 bits physical, 57 bits virtual
Byte Order:                           Little Endian
CPU(s):                               224
On-line CPU(s) list:                  0-223
Vendor ID:                            GenuineIntel
Model name:                           INTEL(R) XEON(R) PLATINUM 8570
CPU family:                           6
Model:                                207
Thread(s) per core:                   2
Core(s) per socket:                   56
Socket(s):                            2
Stepping:                             2
CPU(s) scaling MHz:                   21%
CPU max MHz:                          4000.0000
CPU min MHz:                          800.0000
BogoMIPS:                             4200.00
Flags:                                fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf tsc_known_freq pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_l3 cat_l2 cdp_l3 intel_ppin cdp_l2 ssbd mba ibrs ibpb stibp ibrs_enhanced tpr_shadow flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid cqm rdt_a avx512f avx512dq rdseed adx smap avx512ifma clflushopt clwb intel_pt avx512cd sha_ni avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local split_lock_detect user_shstk avx_vnni avx512_bf16 wbnoinvd dtherm ida arat pln pts hwp hwp_act_window hwp_epp hwp_pkg_req vnmi avx512vbmi umip pku ospke waitpkg avx512_vbmi2 gfni vaes vpclmulqdq avx512_vnni avx512_bitalg tme avx512_vpopcntdq la57 rdpid bus_lock_detect cldemote movdiri movdir64b enqcmd fsrm md_clear serialize tsxldtrk pconfig arch_lbr ibt amx_bf16 avx512_fp16 amx_tile amx_int8 flush_l1d arch_capabilities ibpb_exit_to_user
Virtualization:                       VT-x
L1d cache:                            5.3 MiB (112 instances)
L1i cache:                            3.5 MiB (112 instances)
L2 cache:                             224 MiB (112 instances)
L3 cache:                             600 MiB (2 instances)
NUMA node(s):                         2
NUMA node0 CPU(s):                    0-55,112-167
NUMA node1 CPU(s):                    56-111,168-223
Vulnerability Gather data sampling:   Not affected
Vulnerability Itlb multihit:          Not affected
Vulnerability L1tf:                   Not affected
Vulnerability Mds:                    Not affected
Vulnerability Meltdown:               Not affected
Vulnerability Mmio stale data:        Not affected
Vulnerability Reg file data sampling: Not affected
Vulnerability Retbleed:               Not affected
Vulnerability Spec rstack overflow:   Not affected
Vulnerability Spec store bypass:      Mitigation; Speculative Store Bypass disabled via prctl
Vulnerability Spectre v1:             Mitigation; usercopy/swapgs barriers and __user pointer sanitization
Vulnerability Spectre v2:             Mitigation; Enhanced / Automatic IBRS; IBPB conditional; RSB filling; PBRSB-eIBRS SW sequence; BHI BHI_DIS_S
Vulnerability Srbds:                  Not affected
Vulnerability Tsx async abort:        Not affected
Vulnerability Vmscape:                Mitigation; IBPB before exit to userspace

==============================
Versions of relevant libraries
==============================
[pip3] flashinfer-python==0.6.3
[pip3] mypy==1.11.1
[pip3] mypy_extensions==1.1.0
[pip3] numpy==2.2.6
[pip3] nvidia-cublas-cu12==12.8.4.1
[pip3] nvidia-cuda-cupti-cu12==12.8.90
[pip3] nvidia-cuda-nvrtc-cu12==12.8.93
[pip3] nvidia-cuda-runtime-cu12==12.8.90
[pip3] nvidia-cudnn-cu12==9.10.2.21
[pip3] nvidia-cudnn-frontend==1.18.0
[pip3] nvidia-cufft-cu12==11.3.3.83
[pip3] nvidia-cufile-cu12==1.13.1.3
[pip3] nvidia-curand-cu12==10.3.9.90
[pip3] nvidia-cusolver-cu12==11.7.3.90
[pip3] nvidia-cusparse-cu12==12.5.8.93
[pip3] nvidia-cusparselt-cu12==0.7.1
[pip3] nvidia-cutlass-dsl==4.4.1
[pip3] nvidia-cutlass-dsl-libs-base==4.4.1
[pip3] nvidia-ml-py==13.590.48
[pip3] nvidia-nccl-cu12==2.27.5
[pip3] nvidia-nvjitlink-cu12==12.8.93
[pip3] nvidia-nvshmem-cu12==3.4.5
[pip3] nvidia-nvtx-cu12==12.8.90
[pip3] onnxruntime==1.24.2
[pip3] pyzmq==27.1.0
[pip3] torch==2.10.0
[pip3] torchaudio==2.10.0
[pip3] torchsde==0.2.6
[pip3] torchvision==0.25.0
[pip3] transformers==4.57.6
[pip3] triton==3.6.0
[conda] Could not collect

==============================
         vLLM Info
==============================
ROCM Version                 : Could not collect
vLLM Version                 : 0.16.0
vLLM-Omni Version            : 0.16.1.dev8+g22a51a7cd (git sha: 22a51a7cd)
vLLM Build Flags:
  CUDA Archs: Not Set; ROCm: Disabled
GPU Topology:
        GPU0    GPU1    GPU2    GPU3    GPU4    GPU5    GPU6    GPU7    NIC0    NIC1    NIC2    NIC3    NIC4    NIC5    NIC6    NIC7    NIC8    NIC9    NIC10   NIC11   NIC12NIC13    NIC14   NIC15   CPU Affinity    NUMA Affinity   GPU NUMA ID
GPU0     X      NV18    NV18    NV18    NV18    NV18    NV18    NV18    NODE    NODE    NODE    NODE    PXB     NODE    NODE    NODE    NODE    NODE    SYS     SYS     SYS  SYS      SYS     SYS     0-55,112-167    0               N/A
GPU1    NV18     X      NV18    NV18    NV18    NV18    NV18    NV18    NODE    NODE    NODE    NODE    NODE    NODE    NODE    PXB     NODE    NODE    SYS     SYS     SYS  SYS      SYS     SYS     0-55,112-167    0               N/A
GPU2    NV18    NV18     X      NV18    NV18    NV18    NV18    NV18    NODE    NODE    NODE    NODE    NODE    NODE    NODE    NODE    PXB     NODE    SYS     SYS     SYS  SYS      SYS     SYS     0-55,112-167    0               N/A
GPU3    NV18    NV18    NV18     X      NV18    NV18    NV18    NV18    NODE    NODE    NODE    NODE    NODE    NODE    NODE    NODE    NODE    PXB     SYS     SYS     SYS  SYS      SYS     SYS     0-55,112-167    0               N/A
GPU4    NV18    NV18    NV18    NV18     X      NV18    NV18    NV18    SYS     SYS     SYS     SYS     SYS     SYS     SYS     SYS     SYS     SYS     PXB     NODE    NODE NODE     NODE    NODE    56-111,168-223  1               N/A
GPU5    NV18    NV18    NV18    NV18    NV18     X      NV18    NV18    SYS     SYS     SYS     SYS     SYS     SYS     SYS     SYS     SYS     SYS     NODE    NODE    NODE PXB      NODE    NODE    56-111,168-223  1               N/A
GPU6    NV18    NV18    NV18    NV18    NV18    NV18     X      NV18    SYS     SYS     SYS     SYS     SYS     SYS     SYS     SYS     SYS     SYS     NODE    NODE    NODE NODE     PXB     NODE    56-111,168-223  1               N/A
GPU7    NV18    NV18    NV18    NV18    NV18    NV18    NV18     X      SYS     SYS     SYS     SYS     SYS     SYS     SYS     SYS     SYS     SYS     NODE    NODE    NODE NODE     NODE    PXB     56-111,168-223  1               N/A
NIC0    NODE    NODE    NODE    NODE    SYS     SYS     SYS     SYS      X      PIX     PIX     PIX     NODE    NODE    NODE    NODE    NODE    NODE    SYS     SYS     SYS  SYS      SYS     SYS
NIC1    NODE    NODE    NODE    NODE    SYS     SYS     SYS     SYS     PIX      X      PIX     PIX     NODE    NODE    NODE    NODE    NODE    NODE    SYS     SYS     SYS  SYS      SYS     SYS
NIC2    NODE    NODE    NODE    NODE    SYS     SYS     SYS     SYS     PIX     PIX      X      PIX     NODE    NODE    NODE    NODE    NODE    NODE    SYS     SYS     SYS  SYS      SYS     SYS
NIC3    NODE    NODE    NODE    NODE    SYS     SYS     SYS     SYS     PIX     PIX     PIX      X      NODE    NODE    NODE    NODE    NODE    NODE    SYS     SYS     SYS  SYS      SYS     SYS
NIC4    PXB     NODE    NODE    NODE    SYS     SYS     SYS     SYS     NODE    NODE    NODE    NODE     X      NODE    NODE    NODE    NODE    NODE    SYS     SYS     SYS  SYS      SYS     SYS
NIC5    NODE    NODE    NODE    NODE    SYS     SYS     SYS     SYS     NODE    NODE    NODE    NODE    NODE     X      PIX     NODE    NODE    NODE    SYS     SYS     SYS  SYS      SYS     SYS
NIC6    NODE    NODE    NODE    NODE    SYS     SYS     SYS     SYS     NODE    NODE    NODE    NODE    NODE    PIX      X      NODE    NODE    NODE    SYS     SYS     SYS  SYS      SYS     SYS
NIC7    NODE    PXB     NODE    NODE    SYS     SYS     SYS     SYS     NODE    NODE    NODE    NODE    NODE    NODE    NODE     X      NODE    NODE    SYS     SYS     SYS  SYS      SYS     SYS
NIC8    NODE    NODE    PXB     NODE    SYS     SYS     SYS     SYS     NODE    NODE    NODE    NODE    NODE    NODE    NODE    NODE     X      NODE    SYS     SYS     SYS  SYS      SYS     SYS
NIC9    NODE    NODE    NODE    PXB     SYS     SYS     SYS     SYS     NODE    NODE    NODE    NODE    NODE    NODE    NODE    NODE    NODE     X      SYS     SYS     SYS  SYS      SYS     SYS
NIC10   SYS     SYS     SYS     SYS     PXB     NODE    NODE    NODE    SYS     SYS     SYS     SYS     SYS     SYS     SYS     SYS     SYS     SYS      X      NODE    NODE NODE     NODE    NODE
NIC11   SYS     SYS     SYS     SYS     NODE    NODE    NODE    NODE    SYS     SYS     SYS     SYS     SYS     SYS     SYS     SYS     SYS     SYS     NODE     X      PIX  NODE     NODE    NODE
NIC12   SYS     SYS     SYS     SYS     NODE    NODE    NODE    NODE    SYS     SYS     SYS     SYS     SYS     SYS     SYS     SYS     SYS     SYS     NODE    PIX      X   NODE     NODE    NODE
NIC13   SYS     SYS     SYS     SYS     NODE    PXB     NODE    NODE    SYS     SYS     SYS     SYS     SYS     SYS     SYS     SYS     SYS     SYS     NODE    NODE    NODE  X       NODE    NODE
NIC14   SYS     SYS     SYS     SYS     NODE    NODE    PXB     NODE    SYS     SYS     SYS     SYS     SYS     SYS     SYS     SYS     SYS     SYS     NODE    NODE    NODE NODE      X      NODE
NIC15   SYS     SYS     SYS     SYS     NODE    NODE    NODE    PXB     SYS     SYS     SYS     SYS     SYS     SYS     SYS     SYS     SYS     SYS     NODE    NODE    NODE NODE     NODE     X 

Legend:

  X    = Self
  SYS  = Connection traversing PCIe as well as the SMP interconnect between NUMA nodes (e.g., QPI/UPI)
  NODE = Connection traversing PCIe as well as the interconnect between PCIe Host Bridges within a NUMA node
  PHB  = Connection traversing PCIe as well as a PCIe Host Bridge (typically the CPU)
  PXB  = Connection traversing multiple PCIe bridges (without traversing the PCIe Host Bridge)
  PIX  = Connection traversing at most a single PCIe bridge
  NV#  = Connection traversing a bonded set of # NVLinks

NIC Legend:

  NIC0: mlx5_0
  NIC1: mlx5_1
  NIC2: mlx5_2
  NIC3: mlx5_3
  NIC4: mlx5_4
  NIC5: mlx5_5
  NIC6: mlx5_6
  NIC7: mlx5_7
  NIC8: mlx5_8
  NIC9: mlx5_9
  NIC10: mlx5_10
  NIC11: mlx5_11
  NIC12: mlx5_12
  NIC13: mlx5_13
  NIC14: mlx5_14
  NIC15: mlx5_15

==============================
     Environment Variables
==============================
LD_LIBRARY_PATH=/home/ZhanqiuHu/code/vllm-omni/.venv/lib/python3.12/site-packages/cv2/../../lib64:/usr/local/cuda-13.0/lib64:/usr/local/cuda-13/lib64:/usr/local/cuda-13/lib64
CUDA_HOME=/usr/local/cuda-13
CUDA_HOME=/usr/local/cuda-13
PYTORCH_NVML_BASED_CUDA_CHECK=1
TORCHINDUCTOR_COMPILE_THREADS=1
TORCHINDUCTOR_CACHE_DIR=/tmp/torchinductor_ZhanqiuHu

(vllm-omni) ZhanqiuHu@dgx-b200-01:~/code/vllm-omni (main)$ 
```

</details>


### Your code version

<details>
<summary>The commit id or version of vllm</summary>

```text
0.16.0
```
</details>
<details>
<summary>The commit id or version of vllm-omni</summary>

```text
0.16.1.dev8+g22a51a7cd (commit 22a51a7)
```
</details>

### 🐛 Describe the bug

## Bug Description
Running Qwen3-TTS offline inference (`end2end.py --query-type CustomVoice`) crashes on the first forward pass of the code predictor in Stage-0.

`_maybe_init_kv_cache()` is called before the `set_current_vllm_config()` context is entered, so FlashInfer's constructor fails when it tries to read the global config.

```python
python examples/offline_inference/qwen3_tts/end2end.py --query-type CustomVoice
```

```
(Worker pid=3467599) [Stage-0] ERROR 03-02 20:15:10 [multiproc_executor.py:863] Traceback (most recent call last):
(Worker pid=3467599) [Stage-0] ERROR 03-02 20:15:10 [multiproc_executor.py:863]   File "vllm/v1/executor/multiproc_executor.py", line 858, in worker_busy_loop
(Worker pid=3467599) [Stage-0] ERROR 03-02 20:15:10 [multiproc_executor.py:863]     output = func(*args, **kwargs)
(Worker pid=3467599) [Stage-0] ERROR 03-02 20:15:10 [multiproc_executor.py:863]   File "vllm/v1/worker/worker_base.py", line 361, in execute_model
(Worker pid=3467599) [Stage-0] ERROR 03-02 20:15:10 [multiproc_executor.py:863]     return self.worker.execute_model(scheduler_output)
(Worker pid=3467599) [Stage-0] ERROR 03-02 20:15:10 [multiproc_executor.py:863]   File "vllm/v1/worker/gpu_worker.py", line 652, in execute_model
(Worker pid=3467599) [Stage-0] ERROR 03-02 20:15:10 [multiproc_executor.py:863]     output = self.model_runner.execute_model(
(Worker pid=3467599) [Stage-0] ERROR 03-02 20:15:10 [multiproc_executor.py:863]   File "vllm_omni/worker/gpu_ar_model_runner.py", line 249, in execute_model
(Worker pid=3467599) [Stage-0] ERROR 03-02 20:15:10 [multiproc_executor.py:863]     ) = self._preprocess(scheduler_output, num_tokens_padded, intermediate_tensors)
(Worker pid=3467599) [Stage-0] ERROR 03-02 20:15:10 [multiproc_executor.py:863]   File "vllm_omni/worker/gpu_model_runner.py", line 1208, in _preprocess
(Worker pid=3467599) [Stage-0] ERROR 03-02 20:15:10 [multiproc_executor.py:863]     self._talker_mtp_forward(decode_req_ids, inputs_embeds)
(Worker pid=3467599) [Stage-0] ERROR 03-02 20:15:10 [multiproc_executor.py:863]   File "vllm_omni/worker/gpu_model_runner.py", line 1241, in _talker_mtp_forward
(Worker pid=3467599) [Stage-0] ERROR 03-02 20:15:10 [multiproc_executor.py:863]     req_embeds, code_predictor_codes = self.talker_mtp(...)
(Worker pid=3467599) [Stage-0] ERROR 03-02 20:15:10 [multiproc_executor.py:863]   File "vllm_omni/model_executor/models/qwen3_tts/qwen3_tts_talker.py", line 1568, in talker_mtp
(Worker pid=3467599) [Stage-0] ERROR 03-02 20:15:10 [multiproc_executor.py:863]     audio_codes = self.code_predictor(
(Worker pid=3467599) [Stage-0] ERROR 03-02 20:15:10 [multiproc_executor.py:863]   File "vllm_omni/model_executor/models/qwen3_tts/qwen3_tts_code_predictor_vllm.py", line 492, in forward
(Worker pid=3467599) [Stage-0] ERROR 03-02 20:15:10 [multiproc_executor.py:863]     logits = self.prefill_logits(prefill_input)
(Worker pid=3467599) [Stage-0] ERROR 03-02 20:15:10 [multiproc_executor.py:863]   File "vllm_omni/model_executor/models/qwen3_tts/qwen3_tts_code_predictor_vllm.py", line 393, in prefill_logits
(Worker pid=3467599) [Stage-0] ERROR 03-02 20:15:10 [multiproc_executor.py:863]     self._maybe_init_kv_cache(inputs_embeds.device)
(Worker pid=3467599) [Stage-0] ERROR 03-02 20:15:10 [multiproc_executor.py:863]   File "vllm_omni/model_executor/models/qwen3_tts/qwen3_tts_code_predictor_vllm.py", line 377, in _maybe_init_kv_cache
(Worker pid=3467599) [Stage-0] ERROR 03-02 20:15:10 [multiproc_executor.py:863]     self._kv_cache = _LocalPredictorKVCache(
(Worker pid=3467599) [Stage-0] ERROR 03-02 20:15:10 [multiproc_executor.py:863]   File "vllm_omni/model_executor/models/qwen3_tts/qwen3_tts_code_predictor_vllm.py", line 67, in __init__
(Worker pid=3467599) [Stage-0] ERROR 03-02 20:15:10 [multiproc_executor.py:863]     self.attn_backends, self.attn_metadata_builders = attn_utils.init_attn_backend(
(Worker pid=3467599) [Stage-0] ERROR 03-02 20:15:10 [multiproc_executor.py:863]   File "vllm/v1/worker/gpu/attn_utils.py", line 50, in init_attn_backend
(Worker pid=3467599) [Stage-0] ERROR 03-02 20:15:10 [multiproc_executor.py:863]     attn_metadata_builder = attn_backend.get_builder_cls()(
(Worker pid=3467599) [Stage-0] ERROR 03-02 20:15:10 [multiproc_executor.py:863]   File "vllm/v1/attention/backends/flashinfer.py", line 575, in __init__
(Worker pid=3467599) [Stage-0] ERROR 03-02 20:15:10 [multiproc_executor.py:863]     can_use_trtllm = can_use_trtllm_attention(self.num_qo_heads, self.num_kv_heads)
(Worker pid=3467599) [Stage-0] ERROR 03-02 20:15:10 [multiproc_executor.py:863]   File "vllm/utils/flashinfer.py", line 308, in can_use_trtllm_attention
(Worker pid=3467599) [Stage-0] ERROR 03-02 20:15:10 [multiproc_executor.py:863]     if force_use_trtllm_attention() is False:
(Worker pid=3467599) [Stage-0] ERROR 03-02 20:15:10 [multiproc_executor.py:863]   File "vllm/utils/flashinfer.py", line 302, in force_use_trtllm_attention
(Worker pid=3467599) [Stage-0] ERROR 03-02 20:15:10 [multiproc_executor.py:863]     vllm_config = get_current_vllm_config()
(Worker pid=3467599) [Stage-0] ERROR 03-02 20:15:10 [multiproc_executor.py:863]   File "vllm/config/vllm.py", line 1556, in get_current_vllm_config
(Worker pid=3467599) [Stage-0] ERROR 03-02 20:15:10 [multiproc_executor.py:863]     raise AssertionError(
(Worker pid=3467599) [Stage-0] ERROR 03-02 20:15:10 [multiproc_executor.py:863] AssertionError: Current vLLM config is not set. This typically means get_current_vllm_config() was called outside of a set_current_vllm_config() context, or a CustomOp was instantiated at module import time or model forward time when config is not set. For tests that directly test custom ops/modules, use the 'default_vllm_config' pytest fixture from tests/conftest.py.
```

## Proposed fix:
`_maybe_init_kv_cache()` creates `_LocalPredictorKVCache` which calls `attn_utils.init_attn_backend()` → `FlashInferMetadataBuilder.__init__()` → `get_current_vllm_config()`. This fails because `_maybe_init_kv_cache()` is called in `prefill_logits()` (line 393) *before* the `set_current_vllm_config()` context is entered (line 408).

Fix: wrap the KV cache init in `set_current_vllm_config`:

```python
def _maybe_init_kv_cache(self, device: torch.device) -> None:
    if self._kv_cache is not None:
        return
    with set_current_vllm_config(self._vllm_config):
        # ... existing init code
```

### Before submitting a new issue...

- [x] Make sure you already searched for relevant issues, and asked the chatbot living at the bottom right corner of the [documentation page](https://vllm-omni.readthedocs.io), which can answer lots of frequently asked questions.

--- a/vllm_omni/model_executor/models/qwen3_tts/qwen3_tts_code_predictor_vllm.py
+++ b/vllm_omni/model_executor/models/qwen3_tts/qwen3_tts_code_predictor_vllm.py
@@ -370,16 +370,17 @@ class Qwen3TTSTalkerCodePredictorForConditionalGenerationVLLM(nn.Module):
     def _maybe_init_kv_cache(self, device: torch.device) -> None:
         if self._kv_cache is not None:
             return
-        max_seq_len = int(getattr(self.config, "num_code_groups", 16) or 16)
-        # Upper bound on batch size: vLLM scheduler max_num_seqs (fallback 8).
-        max_batch = int(getattr(self._vllm_config.scheduler_config, "max_num_seqs", 8) or 8)
-        max_batch = max(1, max_batch)
-        self._kv_cache = _LocalPredictorKVCache(
-            vllm_config=self._vllm_config,
-            max_seq_len=max_seq_len,
-            max_batch_size=max_batch,
-            device=device,
-        )
+        with set_current_vllm_config(self._vllm_config):
+            max_seq_len = int(getattr(self.config, "num_code_groups", 16) or 16)
+            # Upper bound on batch size: vLLM scheduler max_num_seqs (fallback 8).
+            max_batch = int(getattr(self._vllm_config.scheduler_config, "max_num_seqs", 8) or 8)
+            max_batch = max(1, max_batch)
+            self._kv_cache = _LocalPredictorKVCache(
+                vllm_config=self._vllm_config,
+                max_seq_len=max_seq_len,
+                max_batch_size=max_batch,
+                device=device,
+            )
 
     @torch.inference_mode()
     def reset_cache(self) -> None:

